firmware/Core/Src/uart_handler/uart_handler.c

```diff
@@ -33,17 +53,49 @@ void HAL_UART_RxCpltCallback(UART_HandleTypeDef *huart)
         UART_telecommand_last_write_time_ms = HAL_GetTick();
         HAL_UART_Receive_IT(&hlpuart1, (uint8_t*) &UART_telecommand_buffer_last_rx_byte, 1);
     }
+    else if (huart->Instance == USART1){
+        // DEBUG_uart_print_str("HAL_UART_RxCpltCallback() -> UART 1 callback called successfully!\n");
+
+        // TODO: Deal with overflow
+        
+        //HAL_UART_DMAPause(&huart1);  // Pause DMA reception to deal with potential interference
+
+        if(current_mpi_mode == COMMAND_MODE) {
+            // char message[50];   // Character array to hold the string
+            // sprintf(message, "MPI response byte: 0x%02X\n", UART_mpi_rx_last_byte);
+            // DEBUG_uart_print_str(".");
+
+            UART_mpi_rx_buffer[UART_mpi_rx_buffer_write_idx++] = UART_mpi_rx_last_byte;
+            UART_mpi_rx_last_byte_write_time_ms = HAL_GetTick();
+        
+            // sprintf(message, "Response Buffer now has byte: 0x%02X\n", UART_mpi_rx_buffer[UART_mpi_rx_buffer_write_idx-1]);
+            // DEBUG_uart_print_str(message);
+
+            //HAL_UART_DMAResume(&huart1);
+            //HAL_UART_Receive_DMA(&huart1, (uint8_t*) &UART_mpi_rx_last_byte, 1);
+        }
+        else{
+            DEBUG_uart_print_str("Unhandled MPI Mode\n"); //FIXME: HANDLE THIS MPI MODE
+        }
+    }
     else {
         // FIXME: add the rest
         DEBUG_uart_print_str("HAL_UART_RxCpltCallback() -> unknown UART instance\n"); // FIXME: remove
     }
 }
 
+// void HAL_UART_RxHalfCpltCallback(UART_HandleTypeDef *huart){
```

Comment:
Please add a clear TODO if leaving commented code in a PR.

Reply:
Part of the outdated code from the rebase errors, have been dealt with in the updated commits.